enumerable.rb

```diff
@@ -0,0 +1,128 @@
+# rubocop:disable Metrics/CyclomaticComplexity
+# rubocop:disable Metrics/PerceivedComplexity
+module Enumerable
+  def my_each
+    return to_enum unless block_given?
+
+    array = to_a
+    array.length.times { |element| yield(array[element]) }
+    self
+  end
+
+  def my_each_with_index
+    return to_enum unless block_given?
+
+    array = to_a
+    array.length.times { |element| yield(array[element], element) }
+    self
+  end
+
+  def my_select
+    return to_enum(:my_select) unless block_given?
+
+    new_arr = []
+    my_each { |it| new_arr << it if yield(it) }
+    new_arr
+  end
+
+  def my_all?(*arg)
+    return "`my_all?': wrong # of arguments (given #{arguments.length}, expected 0..1)" if arguments.length > 1
+
+    if block_given?
+      my_each { |element| return false unless yield(element) }
+    elsif arg[0].is_a? Class
+      my_each { |element| return false unless element.class.ancestors.include?(arg[0]) }
+    elsif arg[0].is_a? Regexp
+      my_each { |element| return false unless arg[0].match(element) }
+    elsif arg.empty?
+      return include?(nil) || include?(false) ? false : true
+    else
+      my_each { |element| return false unless element == arg[0] }
+    end
+    true
+  end
+
+  def my_any?(*arg)
+    return "`my_any?': wrong number of arguments (given #{arguments.length}, expected 0..1)" if arguments.length > 1
+
+    if block_given?
+      my_each { |element| return true if yield(element) }
+    elsif arg.empty?
+      my_each { |element| return true if element }
+      false
+    elsif arg[0].is_a? Class
+      my_each { |element| return true if element.class.ancestors.include?(arg[0]) }
+    elsif arg[0].is_a? Regexp
+      my_each { |element| return true if arg[0].match(element) }
+    else
+      my_each { |element| return true if element == arg[0] }
+    end
+    false
+  end
+
+  def my_none?(*param)
+    return "`my_none': wrong arguments(given #{arguments.length}, expected 0..1)" if arguments.length > 1
+
+    if block_given?
+      my_each { |num| return false if yield(num) }
+    elsif param.empty?
+      my_each { |num| return false unless num.nil? || num == false }
+    elsif param[0].is_a? Regexp
+      my_each { |num| return false if param[0].match(num) }
+    else
+      my_each { |num| return false if num == param[0] }
+    end
+    true
+  end
```

Comment:
- [ ] `my_none` is not working as the original `none?` method, also is not passing all test cases. 

### `none?` is an Enumerable that has these characteristics:
- Returns `true` if all elements are false.
```
[nil, nil, nil].none?
# true
```
- Returns `false` if it finds at least one element that is true
```
[1, 'string', nil].none?
# false
```
- it doesn't change the original array/range/hash
- it can accept as arguments a Class, a RegEx, or a pattern, and returns true or false depending if the values in the array/range/hash match the condition
- it can use a block but it is not necessary to make the method work

[Enumerable#none?](https://ruby-doc.org/core-2.7.2/Enumerable.html#method-i-none-3F)